vault/src/system.rs

```diff
@@ -26,16 +26,25 @@ pub const AUTHORS: &str = env!("CARGO_PKG_AUTHORS");
 pub const NAME: &str = env!("CARGO_PKG_NAME");
 pub const ABOUT: &str = env!("CARGO_PKG_DESCRIPTION");
 
+fn parse_collateral_and_amount(
+    s: &str,
+) -> Result<(CurrencyId, u128), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    let pos = s
+        .find('=')
+        .ok_or_else(|| format!("invalid CurrencyId=amount: no `=` found in `{}`", s))?;
```

Comment:
KEY=value parsing is fairly common IMO and I think it makes sense here since we need to distinguish collateral amounts for each asset. Do you have another suggestion?

Reply:
key=value would have been fine if clap didn't support the --argument=value format already - it's the two potential equal signs in one argument that is bothering me. I guess it'd look ok if you use quotes though, e.g. `--auto-register="ksm=1000"`. An alternative could be `--auto-register="1000 ksm"`.

Reply:
On a related note, if we are making this breaking change anyway, maybe it would be a good moment to switch over to using actual units instead of plancks. I.e., to allow `--auto-register "0.5 ksm"` instead of `500000000` or whatever format we use

Reply:
> it's the two potential equal signs in one argument that is bothering me
That's a fairly common standard in cli parsing and I don't really see a problem with it.

See also: https://github.com/clap-rs/clap/blob/master/examples/keyvalue-derive.rs